ufl/domain.py

```diff
@@ -212,6 +222,10 @@ def ufl_cell(self):
     def is_piecewise_linear_simplex_domain(self):
         return False  # TODO: Any cases this is True
 
+    def ufl_strip_data(self):
+        meshes = (mesh.ufl_strip_data() for mesh in self._ufl_meshes)
+        return TensorProductMesh(meshes, self._ufl_id)
```

Comment:
Is it bad that the ids are preserved for the meshes?

Reply:
I don't know. This was a somewhat arbitrary choice I made to make the two objects as similar as possible.